### PR TITLE
feat/update_gemspec

### PR DESCRIPTION
**Pull request by alChaCC**
update gemspec, readme, add easyship token generator

### CODE REVIEW

File: lib/doorkeeper/jwt_assertion.rb [hunk 1]
```diff
@@ -70,3 +70,13 @@ class ExpiredSignature < DoorkeeperError
     end
   end
 end
+
+module Doorkeeper
+  module Easyship
+    class TokenGenerator
+      def self.generate(options = {})
+        "es_t_#{::Rails.env[0..3]}_" + Base64.strict_encode64(OpenSSL::HMAC.digest(OpenSSL::Digest.new('sha256'), SecureRandom.hex, options.to_s))
```
Comment (alChaCC) [inline lib/doorkeeper/jwt_assertion.rb]: @paulld token will looks like: `es_t_deve_oQj2xl5stDFXA0+dBkxNuHlkOfdiR0Dh2IoEZX2TSdE=`

File: lib/doorkeeper/oauth/assertion_access_token_request.rb [hunk 2]
```diff
@@ -39,114 +39,119 @@ def authorize
...
-            else
-              public_key =  OAuth::Client.find(payload['iss']).application.public_key
-              payload, header = JWT.decode(assertion, OpenSSL::PKey::RSA.new(public_key), true, { algorithm: header['alg'] })
-            end
-          else
-            payload, header = JWT.decode(assertion, configuration.jwt_key)
+        if configuration.jwt_use_application_public_key_as_key
+          payload, header = JWT.decode(assertion, nil, false)
+          raise StandardError.new('App ID is not correct') unless OAuth::Client.find(payload['iss']).present?
+          case header['alg']
```
Comment (alChaCC) [inline lib/doorkeeper/oauth/assertion_access_token_request.rb]: @paulld here is magic that how we support old and new key

File: lib/doorkeeper/oauth/assertion_access_token_request.rb [hunk 3]
```diff
@@ -39,114 +39,119 @@ def authorize
...
+      # @see https://tools.ietf.org/html/draft-ietf-oauth-assertions-18#section-4.1
+      #
+      def create_token
+        client_requested_expires_in = server.jwt['exp'].to_i - server.jwt['iat'].to_i
+        server_expires_in           = Authorization::Token.access_token_expires_in(configuration, client)
+        if server_expires_in
           expires_in = (client_requested_expires_in > 0 && client_requested_expires_in <= server_expires_in) ? client_requested_expires_in : server_expires_in
-          @access_token = AccessToken.find_or_create_for(client, resource_owner.id, scopes, expires_in, configuration.refresh_token_enabled?)
+        else
+          expires_in = nil
```
Comment (alChaCC) [inline lib/doorkeeper/oauth/assertion_access_token_request.rb:145]: @paulld this is how we make token permanent. 
https://github.com/easyship/easyship-api/pull/2921/files#diff-e621685ac1c2689eaedaedf334c6123bR25

File: Gemfile [hunk 4]
```diff
@@ -1,4 +1,4 @@
 source 'https://rubygems.org'
 
-# Specify your gem's dependencies in doorkeeper-jwt_assertion.gemspec
+# Specify your gem's dependencies in doorkeeper-doorkeeoer-jwt.gemspec
```
Comment (paulld) [inline Gemfile]: typo doorkeeper

File: lib/doorkeeper/jwt_assertion.rb (hunk 1 above)
Comment (paulld) [reply lib/doorkeeper/jwt_assertion.rb]: should we just call it  `live_oQj2xl5stDFXA0+dBkxNuHlkOfdiR0Dh2IoEZX2TSdE=`
and `test_oQj2xl5stDFXA0+dBkxNuHlkOfdiR0Dh2IoEZX2TSdE=`?

or  `prod_oQj2xl5stDFXA0+dBkxNuHlkOfdiR0Dh2IoEZX2TSdE=`
and `sand_oQj2xl5stDFXA0+dBkxNuHlkOfdiR0Dh2IoEZX2TSdE=`?

File: lib/doorkeeper/oauth/assertion_access_token_request.rb (hunk 2 above)
Comment (paulld) [reply lib/doorkeeper/oauth/assertion_access_token_request.rb]: is this needed if we don't update the key? do you think that one day we will give the key again, not only the token? to do oauth maybe?

**Review by paulld — COMMENTED** [review]
looks good!

File: Gemfile (hunk 4 above)
Comment (alChaCC) [reply Gemfile]: fixed

File: lib/doorkeeper/jwt_assertion.rb (hunk 1 above)
Comment (alChaCC) [reply lib/doorkeeper/jwt_assertion.rb]: updated to `prod_`

File: lib/doorkeeper/oauth/assertion_access_token_request.rb (hunk 2 above)
Comment (alChaCC) [reply lib/doorkeeper/oauth/assertion_access_token_request.rb]: removed

**Comment by alChaCC** [comment]
@paulld fixed

**Review by paulld — APPROVED** [review]

**Pull request merged**